src/internal/middleware/validation/interceptors.go

```diff
@@ -29,13 +33,22 @@ func UnaryServerInterceptor(ctx context.Context, req any, info *grpc.UnaryServer
 	} else {
 		log.DPanic(ctx, "no validation routine on request message", zap.String("type", fmt.Sprintf("%T", req)))
 	}
-	return handler(ctx, req)
+
+	resp, err := handler(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if b, ok := resp.(branchNillable); ok {
+		b.NilBranch()
```

Comment:
It would be nice if a non-nil branch logged at level dpanic instead.  This would make tests that return non-nil branches fail, and we could fix everything at the source.

It would also be nice if we called ValidateAll on the way out, and implemented the nil branch requirement through validation rules.  (Validation failure on output should just be a log.DPanic.)